### PR TITLE
[libcontacts] Fix unit tests

### PR DESCRIPTION
**Pull request by chriadam**
Commit 813eafc0db80113f79dda9ede2bdc27cd6cf9720 changed how formatted
name (FN) properties from vCards are imported, so that now an import
involving de-duplication can result in a single contact with multiple
nicknames (setNickname() actually adds nicknames if the new nickname
does not match a previously set nickname).

This commit fixes the unit test to take this into account.

### CODE REVIEW

**Comment by chriadam** [comment]
@matthewvogt

**Comment by matthewvogt** [comment]
LGTM

**Comment by thp** [comment]
This can be merged I suppose?

**Comment by chriadam** [comment]
I have been meaning to fix the other unit test breakage in this package: resolve() - when run in non-privileged mode, the unit test fails.  I need to figure out (1) why, (2) how to either fix it or force the test to run in privileged mode.

**Comment by chriadam** [comment]
Ok, should be good now.

File: src/seasidecache.cpp [hunk 1]
```diff
@@ -170,6 +170,9 @@ QMap<QString, QString> managerParameters()
     QMap<QString, QString> rv;
     // Report presence changes independently from other contact changes
     rv.insert(QString::fromLatin1("mergePresenceChanges"), QString::fromLatin1("false"));
+    if (Q_UNLIKELY(!QString::fromUtf8(qgetenv("LIBCONTACTS_TEST_MODE")).isEmpty())) {
```
Comment (matthewvogt) [inline src/seasidecache.cpp]: There's no point in adding Q_UNLIKELY to some test that gets run once, it just makes it harder to read...

File: tests/tst_seasideimport/tst_seasideimport.cpp [hunk 2]
```diff
@@ -468,10 +468,15 @@ void tst_SeasideImport::mergedUid()
     QCOMPARE(name.isEmpty(), true);
 
     const QList<QContactNickname> nicknames(c.details<QContactNickname>());
-    QCOMPARE(nicknames.count(), 1);
+    QCOMPARE(nicknames.count(), 2);
```
Comment (matthewvogt) [inline tests/tst_seasideimport/tst_seasideimport.cpp]: I'd prefer this change to be in a separate commit to the other parts.

**Comment by matthewvogt** [comment]
LGTM
